expression/builtin_json.go

```diff
@@ -716,7 +720,70 @@ type jsonValidFunctionClass struct {
 }
 
 func (c *jsonValidFunctionClass) getFunction(ctx sessionctx.Context, args []Expression) (builtinFunc, error) {
-	return nil, errFunctionNotExists.GenWithStackByArgs("FUNCTION", "JSON_VALID")
+	if err := c.verifyArgs(args); err != nil {
+		return nil, err
+	}
+
+	var sig builtinFunc
+	argType := args[0].GetType().EvalType()
+	switch argType {
+	case types.ETJson:
+		bf := newBaseBuiltinFuncWithTp(ctx, args, types.ETInt, types.ETJson)
+		sig = &builtinJSONValidJSONSig{bf}
+		sig.setPbCode(tipb.ScalarFuncSig_JsonValidJsonSig)
+	default:
+		bf := newBaseBuiltinFuncWithTp(ctx, args, types.ETInt, argType)
```

Comment:
```suggestion
		bf := newBaseBuiltinFuncWithTp(ctx, args, types.ETInt, types.ETString)
```

Reply:
If its argument is not a string or a json, should we convert it to a string?

Reply:
In mysql5.7
```
mysql> select json_valid(0);
+---------------+
| json_valid(0) |
+---------------+
|             0 |
+---------------+
1 row in set (0.00 sec)

mysql> select json_valid("0");
+-----------------+
| json_valid("0") |
+-----------------+
|               1 |
+-----------------+
1 row in set (0.00 sec)
```

Reply:
Should we handle int/double/decimal values separately?

Reply:
See [here](https://github.com/mysql/mysql-server/blob/8.0/sql/item_json_func.cc#L295-L319) for more details.